### PR TITLE
enhancement(pkg/scale): remove sorting logic from encodeMap method

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Changes

- Removed sorting logic for map keys in encodeMap method as @timwu20 suggested over [here](https://github.com/ChainSafe/gossamer/pull/2894#discussion_r1005921855) in PR #2894 .

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Primary Reviewer

@timwu20

### CODE REVIEW

File: pkg/scale/encode_test.go [hunk 1]
```diff
@@ -1127,8 +1134,16 @@ func Test_encodeState_encodeMap(t *testing.T) {
...
+			gotExpectedEncode := false
+			for _, expectedValue := range tt.want {
+				if reflect.DeepEqual(buffer.Bytes(), expectedValue) {
+					gotExpectedEncode = true
+					break
+				}
+			}
+			if !gotExpectedEncode {
+				t.Errorf("encodeState.encodeMap() = %v, want %v", buffer.Bytes(), tt.want[0])
 			}
```
Comment (qdm12) [inline pkg/scale/encode_test.go:1147]: nit you could use https://github.com/stretchr/testify/tree/master/assert with

```suggestion
			assert.Contains(t, tt.want, buffer.Bytes())
```

File: pkg/scale/encode_test.go [hunk 2]
```diff
@@ -1116,6 +1097,32 @@ func Test_encodeState_encodeArray(t *testing.T) {
 }
 
 func Test_encodeState_encodeMap(t *testing.T) {
+	mapTests := []struct {
+		name    string
+		in      interface{}
+		wantErr bool
+		want    [][]byte
```
Comment (qdm12) [inline pkg/scale/encode_test.go:1104]: nit maybe rename `want` to `wantOneOf` to avoid any confusion?

**Review by qdm12 — APPROVED** [review]
Nice :100:

**Review by qdm12 — APPROVED** [review]
Tiny FYI, mocks CI job may fail due to https://github.com/ChainSafe/gossamer/pull/2910 you might want to rebase your branch on development once this is merged.

**Review by timwu20 — APPROVED** [review]

**Comment by qdm12** [comment]
@axaysagathiya feel free to squash and merge (if you are allowed to? :thinking:)

**Comment by kishansagathiya** [comment]
He doesn't have permissions to merge. I can merge it.

**Pull request merged**